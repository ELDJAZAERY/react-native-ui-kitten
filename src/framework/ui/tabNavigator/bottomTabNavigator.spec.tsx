import React from 'react';
import { View } from 'react-native';
import {
  fireEvent,
  render,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import * as config from './bottomTabNavigator.spec.config';
import {
  BottomTabNavigator as BottomTabNavigatorComponent,
  Props as BottomTabNavigatorProps,
  TabRoute,
  TAB_TEST_ID,
} from './bottomTabNavigator.component';

const BottomTabNavigator = styled<BottomTabNavigatorComponent, BottomTabNavigatorProps>(BottomTabNavigatorComponent);

describe('@bottom-tab-navigator: component checks', () => {

  const Mock = (props?: BottomTabNavigatorProps): React.ReactElement<StyleProviderProps> => (
    <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
      <BottomTabNavigator {...props} />
    </StyleProvider>
  );

  const routes: TabRoute[] = [
    { routeName: 'Screen1', routeImage: <View style={{width: 20, height: 20, backgroundColor: 'red'}}/> },
    { routeName: 'Screen2', routeImage: <View style={{width: 20, height: 20, backgroundColor: 'red'}}/> },
    { routeName: 'Screen3', routeImage: <View style={{width: 20, height: 20, backgroundColor: 'red'}}/> },
  ];

  it('* empty', () => {
    const component = render(<Mock routes={[]}/>);
    expect(component).toMatchSnapshot();
  });

  it('* with routes', () => {
    const component = render(<Mock routes={routes}/>);
    expect(component).toMatchSnapshot();
  });

  it('* current index', () => {
    const component = render(<Mock routes={routes} currentIndex={1}/>);
    expect(component).toMatchSnapshot();
  });

  it('* tab choose', () => {
    const onTabChoose = jest.fn();
    const component = render(<Mock routes={routes} onTabChoose={onTabChoose}/>);
    fireEvent.press(component.getByTestId(TAB_TEST_ID + '1'));
    expect(component).toMatchSnapshot();
    expect(onTabChoose).toHaveBeenCalled();
  });

  it('* different appearances', () => {
    const component1 = render(<Mock routes={routes} appearance='text'/>);
    const component2 = render(<Mock routes={routes} appearance='highlight'/>);
    const component3 = render(<Mock routes={routes} appearance='text-highlight'/>);
    expect(component1).toMatchSnapshot();
    expect(component2).toMatchSnapshot();
    expect(component3).toMatchSnapshot();
  });

});

