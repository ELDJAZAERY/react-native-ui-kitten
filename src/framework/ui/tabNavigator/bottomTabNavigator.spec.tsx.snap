// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@bottom-tab-navigator: component checks * current index 1`] = `
<View
  appearance="default"
  currentIndex={1}
  dispatch={[Function]}
  routes={
    Array [
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen1",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen2",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen3",
      },
    ]
  }
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": false,
      "showText": false,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-0"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-1"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-2"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * different appearances 1`] = `
<View
  appearance="text"
  currentIndex={0}
  dispatch={[Function]}
  routes={
    Array [
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen1",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen2",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen3",
      },
    ]
  }
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": false,
      "showText": true,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-0"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
        }
      }
    >
      Screen1
    </Text>
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-1"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
        }
      }
    >
      Screen2
    </Text>
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-2"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
        }
      }
    >
      Screen3
    </Text>
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * different appearances 2`] = `
<View
  appearance="highlight"
  currentIndex={0}
  dispatch={[Function]}
  routes={
    Array [
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen1",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen2",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen3",
      },
    ]
  }
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": true,
      "showText": false,
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={0}
    style={
      Object {
        "0": Object {
          "position": "absolute",
        },
        "1": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-0"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-1"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-2"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * different appearances 3`] = `
<View
  appearance="text-highlight"
  currentIndex={0}
  dispatch={[Function]}
  routes={
    Array [
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen1",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen2",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen3",
      },
    ]
  }
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": true,
      "showText": true,
    }
  }
>
  <View
    animationDuration={200}
    onLayout={[Function]}
    positions={3}
    selectedPosition={0}
    style={
      Object {
        "0": Object {
          "position": "absolute",
        },
        "1": Object {
          "backgroundColor": "#3366FF",
          "height": 3,
        },
        "backgroundColor": "#3366FF",
        "height": 3,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "33.333333333333336%",
      }
    }
  />
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-0"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3366FF",
        }
      }
    >
      Screen1
    </Text>
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-1"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
        }
      }
    >
      Screen2
    </Text>
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-2"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#8992A3",
        }
      }
    >
      Screen3
    </Text>
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * empty 1`] = `
<View
  appearance="default"
  currentIndex={0}
  dispatch={[Function]}
  routes={Array []}
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": false,
      "showText": false,
    }
  }
/>
`;

exports[`@bottom-tab-navigator: component checks * tab choose 1`] = `
<View
  appearance="default"
  currentIndex={0}
  dispatch={[Function]}
  onTabChoose={
    [MockFunction] {
      "calls": Array [
        Array [
          "Screen2",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  routes={
    Array [
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen1",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen2",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen3",
      },
    ]
  }
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": false,
      "showText": false,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-0"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-1"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-2"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`@bottom-tab-navigator: component checks * with routes 1`] = `
<View
  appearance="default"
  currentIndex={0}
  dispatch={[Function]}
  routes={
    Array [
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen1",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen2",
      },
      Object {
        "routeImage": <View
          style={
            Object {
              "backgroundColor": "red",
              "height": 20,
              "width": 20,
            }
          }
        />,
        "routeName": "Screen3",
      },
    ]
  }
  style={
    Array [
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": 750,
      },
      Object {
        "backgroundColor": "transparent",
        "borderTopColor": "#A6AEBD",
        "borderTopWidth": 1,
        "paddingVertical": 16,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "transparent",
      "borderTopColor": "#A6AEBD",
      "borderTopWidth": 1,
      "color": "#8992A3",
      "highlightHeight": 3,
      "paddingVertical": 16,
      "selectedColor": "#3366FF",
      "showHighlight": false,
      "showText": false,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-0"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-1"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    testID="@tab-navigator/tab-2"
  >
    <View
      style={
        Object {
          "backgroundColor": "red",
          "height": 20,
          "marginBottom": 5,
          "width": 20,
        }
      }
    />
  </View>
</View>
`;
